src/tests/JIT/Directed/tailcall/mutual_recursion.fs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// Notes:
+//
+// This test will Stack overflow when built in debug. It relies on the FSharp
+// compiler optimizing the tail calls to loops.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+open System
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+// This is a 16 byte struct
+type Point2D = struct
+    val X: double
+    val Y: double
+
+    new(x: double, y:double) = { X = x; Y = y}
+
+    end
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+type DriverMethods() =
+    static member FirstMethodCallee(firstArg: int, secondArg: int, thirdArg: int) = 
+        if firstArg = 10 then firstArg * secondArg * thirdArg
+        else firstArg + secondArg + thirdArg
+
+    static member FirstMethod() = 
+        let firstArg = 10
+        let secondArg = 20
+        let thirdArg = 30
+
+        DriverMethods.FirstMethodCallee(firstArg, secondArg, thirdArg)
+
+    static member SecondMethodCallee(iteration: int, firstArg: int, secondArg: int, thirdArg: int) = 
+        if iteration = 0 then
+            firstArg
+        else
+            let mutable retVal = 0
+            if firstArg = 10 then 
+                retVal <- firstArg * secondArg * thirdArg
+            else 
+                retVal <- firstArg + secondArg + thirdArg
```

Comment:
```suggestion
            let mutable retVal =
                if firstArg = 10 then 
                    firstArg * secondArg * thirdArg
                else 
                    firstArg + secondArg + thirdArg
```
Typically in F# programs we try to make things expressions unless mutation is strictly more efficient